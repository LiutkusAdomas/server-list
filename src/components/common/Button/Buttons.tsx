import { MouseEventHandler, ReactNode } from 'react';

type ButtonPropsType = {
  onClick: MouseEventHandler<HTMLButtonElement>;
  label?: string;
  disabled?: boolean;
  children?: ReactNode;
};

const Button: React.FC<ButtonPropsType> = ({ label, onClick, disabled, children }) => {
  return (
    <button
      disabled={disabled}
      onClick={onClick}
      className={`group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md shadow-md text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500 ${
        disabled ? 'bg-gray-400 cursor-not-allowed hover:bg-gray-400' : ''
      }`}
    >
      {label}
      {children}
    </button>
  );
};

export default Button;
