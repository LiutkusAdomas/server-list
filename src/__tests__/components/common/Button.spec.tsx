import { render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import { Button } from '../../../components/common/Button/Buttons';

describe('Button component', () => {
  it('should render a button with default props', () => {
    const { asFragment } = render(<Button onClick={vi.fn()}> </Button>);
    expect(asFragment).toMatchSnapshot();
  });
  it('should render a button with a label', () => {
    const { asFragment } = render(<Button onClick={vi.fn()} label="Test"></Button>);
    expect(asFragment).toMatchSnapshot();
  });
  it('should render a disabled button', () => {
    const { asFragment } = render(<Button onClick={vi.fn()} disabled={true}></Button>);
    expect(asFragment).toMatchSnapshot();
  });
  it('should render a non disabled button', () => {
    const { asFragment } = render(<Button onClick={vi.fn()} disabled={false}></Button>);
    expect(asFragment).toMatchSnapshot();
  });
  it('should render a button with some text inside', () => {
    const { asFragment } = render(<Button onClick={vi.fn()}>There is some text</Button>);
    expect(asFragment).toMatchSnapshot();
  });
});
