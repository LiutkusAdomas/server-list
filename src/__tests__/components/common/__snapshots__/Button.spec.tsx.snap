// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > should render a button with a label 1`] = `[Function]`;

exports[`Button component > should render a button with default props 1`] = `[Function]`;

exports[`Button component > should render a button with some text inside 1`] = `[Function]`;

exports[`Button component > should render a disabled button 1`] = `[Function]`;

exports[`Button component > should render a non disabled button 1`] = `[Function]`;
